### PR TITLE
Feature/wf 42 e button

### PR DESCRIPTION
**Pull request by patric-eberle**
## Pull request
This pull request adds a e-button component and an e-process component. I created a example page so you can test in all browsers.

Please note: because of some reset styles of vue-styleguidist, the styles are not 100% valid in the component styleguide (e.g. buttons have a black font on iOS Safari). The demo page shows the correct styles.

Please note: the process animation fails in Edge. I will check this with Christoph at a later stage to find an alternative.
 
### Ticket
https://winterhalter-fenner.atlassian.net/browse/WF-446
 
### Browser testing
- http://localhost:6060/styleguide#!/e-button
- http://localhost:8080/styleguide/buttons
 
### Checklist
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [x] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [x] Did run automated tests and linters
- [x] Did review code
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Re-assign ticket to developer

### CODE REVIEW

File: app/components/e-progress.vue [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      <div :class="b('bubble')"></div>
+      <div :class="b('bubble')"></div>
+      <div :class="b('bubble')"></div>
+    </div>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'e-progress', // TODO: update name according to filename
```
Comment (geroldboehler) [inline app/components/e-progress.vue]: Remove TODO

File: app/components/e-progress.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+### Styles
```
Comment (geroldboehler) [inline app/components/e-progress.md]: e-progress is not rendering in styleguide.

File: app/components/e-progress.vue [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    // mixins: [],
+
+    props: {
+      negative: {
+        type: Boolean,
+        default: false
+      },
+
+      spacing: {
+        type: [String, Number],
```
Comment (geroldboehler) [inline app/components/e-progress.vue]: Number's are not working with vue-bem-cn currently.

File: app/components/e-progress.vue (hunk 1 above)
Comment (patric-eberle) [reply app/components/e-progress.vue]: Done.

File: app/components/e-progress.md (hunk 2 above)
Comment (patric-eberle) [reply app/components/e-progress.md]: This is caused by an invalid vue-styleguidist version, which did not support mixins. Please reinstall node packages while in branch and check again.

**Comment by patric-eberle** [comment]
@geroldboehler fixed issues and IE11 Bug. Please review again.

**Comment by geroldboehler** [comment]
Looks good to me, the only thing i noticed is that the progress animation is rendered wrong in IE11. Otherwise ok to merge.

**Pull request merged**